openbas-api/src/main/java/io/openbas/rest/injector_contract/service/InjectorContractService.java

```diff
@@ -0,0 +1,117 @@
+package io.openbas.rest.injector_contract.service;
+
+import io.openbas.database.model.Filters;
+import io.openbas.database.model.InjectorContract;
+import io.openbas.database.raw.RawInjectorsContrats;
+import io.openbas.database.repository.AttackPatternRepository;
+import io.openbas.database.repository.InjectorContractRepository;
+import io.openbas.database.repository.InjectorRepository;
+import io.openbas.database.specification.InjectorContractSpecification;
+import io.openbas.rest.exception.ElementNotFoundException;
+import io.openbas.rest.injector_contract.form.InjectorContractAddInput;
+import io.openbas.rest.injector_contract.form.InjectorContractUpdateInput;
+import io.openbas.rest.injector_contract.form.InjectorContractUpdateMappingInput;
+import io.openbas.utils.pagination.SearchPaginationInput;
+import jakarta.transaction.Transactional;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.List;
+
+import static io.openbas.helper.DatabaseHelper.updateRelation;
+import static io.openbas.helper.StreamHelper.fromIterable;
+import static io.openbas.utils.pagination.PaginationUtils.buildPaginationJPA;
+
+@Service
+@RequiredArgsConstructor
+public class InjectorContractService {
```

Comment:
I get what you want to do here by migrating functional into a service layer but I don't think we should do such impactful changes during a bug fix.
Besides, since we have a team discussion about refactoring the structure of the project, we should wait for it before doing anything on that part.

Reply:
I think we should start with a migration as small as that. I also think we all agree that we need to make these changes. But you're also right, it can be very impactful. I did the rollback